cephfs/cephfs_test.go

```diff
@@ -42,164 +50,169 @@ func TestSyncFs(t *testing.T) {
 	err = mount.Mount()
 	assert.NoError(t, err)
 
-	err = mount.SyncFs()
+	err = mount.Unmount()
 	assert.NoError(t, err)
+	assert.False(t, mount.IsMounted())
 }
 
-func TestChangeDir(t *testing.T) {
+// TestReleaseMount tests releasing a ceph mount handle
+func TestReleaseMount(t *testing.T) {
 	mount, err := cephfs.CreateMount()
 	assert.NoError(t, err)
 	assert.NotNil(t, mount)
 
-	err = mount.ReadDefaultConfigFile()
-	assert.NoError(t, err)
-
-	err = mount.Mount()
-	assert.NoError(t, err)
-
-	dir1 := mount.CurrentDir()
-	assert.NotNil(t, dir1)
-
-	err = mount.MakeDir("/asdf", 0755)
-	assert.NoError(t, err)
-
-	err = mount.ChangeDir("/asdf")
+	err = mount.Release()
 	assert.NoError(t, err)
-
-	dir2 := mount.CurrentDir()
-	assert.NotNil(t, dir2)
-
-	assert.NotEqual(t, dir1, dir2)
-	assert.Equal(t, dir1, "/")
-	assert.Equal(t, dir2, "/asdf")
 }
 
-func TestRemoveDir(t *testing.T) {
-	dirname := "one"
-	mount, err := cephfs.CreateMount()
+// TestSyncFs tests syncing filesystem data to persistent media
+func TestSyncFs(t *testing.T) {
+	err := withCephMount(func(mount *cephfs.MountInfo) error {
+		err := mount.SyncFs()
+		assert.NoError(t, err)
+		return nil
+	})
 	assert.NoError(t, err)
-	assert.NotNil(t, mount)
+}
 
-	err = mount.ReadDefaultConfigFile()
-	assert.NoError(t, err)
+// TestChangeDir tests changing ceph current working directory
+func TestChangeDir(t *testing.T) {
+	err := withCephMount(func(mount *cephfs.MountInfo) error {
 
-	err = mount.Mount()
-	assert.NoError(t, err)
+		dir1 := mount.CurrentDir()
```

Comment:
this indention looks off. did you run go fmt?

Reply:
Indentation is correct, i have `go fmt` built into my editor.

This is the due to the `withCephMount` usage as defining the function scope + indentation.

Reply:
got it. the way the diffing works for the changes to the test case make it very difficult to review. if you are primarily adding a new code, you might be able to get the diff to look cleaner by adding all new code at the bottom of the file. otherwise, some sufficient comments to describe things or creating a patch that removes and then a patch that adds might make it easier.

Reply:
Okay, TODO: Split into 3 commits, 1 commit with new open/close commit, 1 commit with old testing code with open/close, and 1 commit with withCephMount.

Reply:
mm, i'm even fine keeping everything in a single commit... my only ask is some restructuring that makes the testing code diff easier to review.